scylla/tests/integration/compression.rs

```diff
@@ -0,0 +1,114 @@
+use crate::utils::{test_with_3_node_cluster, unique_keyspace_name, PerformDDL};
+use scylla::client::session::Session;
+use scylla::client::session_builder::SessionBuilder;
+use scylla::statement::query::Query;
+use scylla::frame::Compression;
+
+use scylla_proxy::{ProxyError, WorkerError, ShardAwareness, RequestRule, Reaction, RequestReaction, Condition, RequestOpcode};
+
+use tokio::sync::mpsc;
+use std::sync::Arc;
+
+async fn test_compression(compression: Option<Compression>, text_size: usize, expected_frame_total_size_range: std::ops::Range<usize>) {
+    let res = test_with_3_node_cluster(ShardAwareness::QueryNode, |proxy_uris, translation_map, mut running_proxy| async move {
+    
+        let request_rule = |tx| {
+            RequestRule(
+                Condition::or(Condition::RequestOpcode(RequestOpcode::Execute), Condition::RequestOpcode(RequestOpcode::Query)),
+                RequestReaction::noop().with_feedback_when_performed(tx),
+            )
+        };
```

Comment:
:question: Isn't it erroneous that you don't filter out the control connection? Depending on when the driver issues metadata queries on it and what responses it gets, the sum of frames' lengths can vary.

Reply:
I think that no all of the control parts are getting compressed, so they are less relevant

I can include everything back again, but then we'll need to re-calibrate the possible ranges of data length